### PR TITLE
added connection to PostgreSQL bd, realisation of ItemDao with JDBC

### PR DESCRIPTION
**Pull request by SkaterWeather**

### CODE REVIEW

File: src/main/java/internetshop/Factory.java [hunk 1]
```diff
@@ -28,10 +47,7 @@
     private static UserService userServiceInstance;
 
     public static ItemDao getItemDao() {
-        if (itemDaoInstance == null) {
-            itemDaoInstance = new ItemDaoImpl();
```
Comment (Sofasmile) [inline src/main/java/internetshop/Factory.java:32]: if (itemDaoInstance == null) {
            itemDaoInstance = new ItemDaoJdbcImpl(connection);
        }
        return itemDaoInstance;

File: src/main/java/internetshop/controller/inits/DataInitializer.java [hunk 2]
```diff
@@ -27,29 +27,25 @@
 
     @Override
     public void contextInitialized(ServletContextEvent servletContextEvent) {
-        Item firstItem = new Item("firstItem", 9.54);
-        Item secondItem = new Item("secondItem", 3.48);
-        Item thirdItem = new Item("thirdItem", 10.00);
-        itemService.create(firstItem);
-        itemService.create(secondItem);
-        itemService.create(thirdItem);
+        //testing itemDaoJdbcImpl.class
```
Comment (Sofasmile) [inline src/main/java/internetshop/controller/inits/DataInitializer.java]: remove commented code

File: src/main/java/internetshop/dao/jdbc/ItemDaoJdbcImpl.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    public Item create(Item item) {
+        String query = String.format("INSERT INTO %s (id, name, price) VALUES (%d, '%s', %f);",
+                ITEMS_TABLE, item.getId(), item.getName(), item.getPrice());
+
+        try {
+            Statement statement = connection.createStatement();
+            statement.executeUpdate(query);
+        } catch (SQLException e) {
+            logger.warn("Can't create item " + item);
+        }
```
Comment (Sofasmile) [inline src/main/java/internetshop/dao/jdbc/ItemDaoJdbcImpl.java:33]: add block finaly

File: src/main/java/internetshop/dao/jdbc/ItemDaoJdbcImpl.java [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    @Override
+    public Item create(Item item) {
+        String query = String.format("INSERT INTO %s (id, name, price) VALUES (%d, '%s', %f);",
+                ITEMS_TABLE, item.getId(), item.getName(), item.getPrice());
+
+        try {
+            Statement statement = connection.createStatement();
+            statement.executeUpdate(query);
+        } catch (SQLException e) {
+            logger.warn("Can't create item " + item);
```
Comment (Sofasmile) [inline src/main/java/internetshop/dao/jdbc/ItemDaoJdbcImpl.java]:
```suggestion
            logger.error("Can't create item " + item);
```

**Review by Sofasmile — COMMENTED** [review]
make small fixes

File: src/main/java/internetshop/dao/jdbc/ItemDaoJdbcImpl.java (hunk 3 above)
Comment (SkaterWeather) [reply src/main/java/internetshop/dao/jdbc/ItemDaoJdbcImpl.java:33]: then i will use try with resourses

File: src/main/java/internetshop/Factory.java (hunk 1 above)
Comment (SkaterWeather) [reply src/main/java/internetshop/Factory.java:32]: true, it's still should be a singleton

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**